icu4j/main/classes/core/src/com/ibm/icu/text/SimpleDateFormat.java

```diff
@@ -3609,8 +3609,8 @@ private int subParse(String text, int start, char ch, int count,
                 return ~start;
             }
             case 27: // 'Q' - QUARTER
-                if (count <= 2 || (number != null && getBooleanAttribute(DateFormat.BooleanAttribute.PARSE_ALLOW_NUMERIC))) {
-                    // i.e., Q or QQ. or lenient & have number
+                if (count <= 2 && number != null) {
+                    // i.e., Q or QQ.
```

Comment:
I see, you are handling the PARSE_ALLOW_NUMERIC case lower down

Reply:
Exactly.